milestone2.ts

```diff
@@ -139,9 +358,39 @@ type UpdateVolunteer = (
   vol: Volunteer,
   updates: Partial<Volunteer>
 ) => Volunteer;
-let updateVolunteer: UpdateVolunteer; // Code here
+let updateVolunteer: UpdateVolunteer = (vol, updates) =>{
+  //const ret = {...vol, email: updates.email, city:updates.city, state: updates.state};
+  const myName = "name";
+  let volr;
+  if(myName in updates){
+    volr = {name: updates.name, ...vol};
+    return volr;
+  }
```

Comment:
This question is actually a bit simpler than you made it, but I appreciate the effort! All you have to do is spread updates in the same object as vol, and all of the values from update will overwrite the ones for vol. It should look like {...vol, ...updates}